src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -478,6 +461,59 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
         });
   }
 
+  /**
+   * Redirects to login page if feedback session is accessible.
+   *
+   * @param auth The authentication information used in the redirection URL.
+   */
+  redirectToLoginPageIfFeedbackSessionAccessible(auth: AuthInfo): void {
+    this.isFeedbackSessionLoading = true;
+    this.feedbackSessionsService.getFeedbackSession({
+      courseId: this.courseId,
+      feedbackSessionName: this.feedbackSessionName,
+      intent: this.intent,
+      key: this.regKey,
+      moderatedPerson: this.moderatedPerson,
+      previewAs: this.previewAsPerson,
+    }).pipe(finalize(() => {
+      this.isFeedbackSessionLoading = false;
+    }))
+      .subscribe(() => {}, (resp: ErrorMessageOutput) => {
```

Comment:
- Will there be a happy path here? It doesn't seem right that the happy path is swallowed.
- In fact, if you're sure that the happy path will never be reached, why not reuse `loadFeedbackSession`? Then we don't even need to extract out `handleGetFeedbackSessionError` although we still need to modify the error-handling part slightly.